qiskit/pulse/commands/sample_pulse.py

```diff
@@ -31,28 +31,61 @@ class SamplePulse(Command):
 
     prefix = 'p'
 
-    def __init__(self, samples: Union[np.ndarray, List[complex]], name: Optional[str] = None):
+    def __init__(self, samples: Union[np.ndarray, List[complex]], name: Optional[str] = None,
+                 epsilon: float = 1e-6):
         """Create new sample pulse command.
 
         Args:
             samples: Complex array of pulse envelope
             name: Unique name to identify the pulse
-        Raises:
-            PulseError: when pulse envelope amplitude exceeds 1
+            epsilon: Pulse sample norm tolerance for clipping.
+                If any sample's norm exceeds unity by less than or equal to epsilon
+                it will be clipped by scaling its norm by 1-epsilon. If the sample
+                norm is greater than 1+epsilon an error will be raised
         """
         super().__init__(duration=len(samples))
 
-        if np.any(np.abs(samples) > 1):
-            raise PulseError('Absolute value of pulse envelope amplitude exceeds 1.')
+        samples = np.asarray(samples, dtype=np.complex_)
 
-        self._samples = np.asarray(samples, dtype=np.complex_)
+        self._samples = self._clip(samples, epsilon=epsilon)
         self._name = SamplePulse.create_name(name)
 
     @property
     def samples(self):
         """Return sample values."""
         return self._samples
 
+    def _clip(self, samples: np.ndarray, epsilon: float = 1e-6):
+        """If samples are within epsilon of unit norm, clip sample by reducing norm by (1-epsilon).
+
+        If difference is greater than epsilon error is raised.
+
+        Args:
+            samples: Complex array of pulse envelope
+            epsilon: Pulse sample norm tolerance for clipping.
+                If any sample's norm exceeds unity by less than or equal to epsilon
+                it will be clipped by scaling its norm by 1-epsilon. If the sample
+                norm is greater than 1+epsilon an error will be raised
+
+        Returns:
+            np.ndarray: Clipped pulse samples
+        Raises:
+            PulseError: If there exists a pulse sample with a norm greater than 1+epsilon
+        """
+        samples_norm = np.abs(samples)
+        to_clip = (samples_norm > 1.) & (samples_norm <= 1. + epsilon)
+
+        if np.any(to_clip):
+            clip_where = np.argwhere(to_clip)
+            clipped_samples = samples[clip_where]*(1-epsilon)
+            samples[clip_where] = clipped_samples
+            samples_norm[clip_where] = np.abs(clipped_samples)
+
+        if np.any(samples_norm > 1.):
```

Comment:
This check could be done earlier, just after you first define samples_norm  to save some computation.

Reply:
The check needs to be done here since samples that would later be clipped would trigger this exception prior to clipping.